### ISSUE TITLE
Plugin doesn't seem to work

### ISSUE THREAD

**Issue by ryantam626** [ryantam626/jupyterlab_code_formatter#111]
Copy and pasted from issue #110

Quote begins
------------------- 

I am using it for the first time, and in JLab 2.0, are you sure it works? I am trying it with a cell like this:

```python
a='hello'
```
and expect it to be reformatted to:
```python
a = "hello"
```
using black (confirmed in the settings) and black is installed.

I tried Right-click->Format cell, while
* having cell active in editing mode
* having cell selected in command mode
* having the text in the cell selected using the mouse and using the command then.

to no avail.

Edit: Format notebook also doesn't do anything. no error messages in the log.

_Originally posted by @michaelaye in https://github.com/ryantam626/jupyterlab_code_formatter/issues/108#issuecomment-595959264_

**Comment by ryantam626** [comment]
Hi @michaelaye! I made a new issue here just for visibility.

That's odd, I definitely tested that functionaility :sweat: Could you post output of the following commnds please?

`pip freeze`
`jupyter labextension list`
`jupyter serverextension list`

**Comment by Gonzalo933** [comment]
I am experiencing the same bug. This is the output of the commands. (I am doing pip show instead of freeze because I have hundreds of packages)

```
$ pip show jupyterlab_code_formatter
Name: jupyterlab-code-formatter
Version: 1.2.0
Summary: Server extension to power `@ryantam626/jupyterlab_code_formatter` npm package.
Home-page: https://github.com/ryantam626/jupyterlab_code_formatter
Author: Ryan Tam
Author-email: ryantam626@gmail.com
License: MIT
Location: /home/gonzalo.hernandez/anaconda3/lib/python3.7/site-packages
Requires: notebook, packaging
Required-by:
```

```
$  jupyter labextension list
JupyterLab v1.0.2
Known labextensions:
   app dir: /home/gonzalo.hernandez/anaconda3/share/jupyter/lab
        @ryantam626/jupyterlab_code_formatter v1.1.0  enabled  OK
```

```
$  jupyter serverextension list
config dir: /home/gonzalo.hernandez/.jupyter
    jupyter_nbextensions_configurator  enabled
    - Validating...
      jupyter_nbextensions_configurator 0.4.1 OK
    jupyterlab_code_formatter  enabled
    - Validating...
      jupyterlab_code_formatter  OK
config dir: /home/gonzalo.hernandez/anaconda3/etc/jupyter
    jupyterlab  enabled
    - Validating...
      jupyterlab 1.0.2 OK
    jupyter_nbextensions_configurator  enabled
    - Validating...
      jupyter_nbextensions_configurator 0.4.1 OK
```

**Comment by Gonzalo933** [comment]
And the output of the log when doing a code formatting

```
[W 10:21:59.367 LabApp] 422 POST /jupyterlab_code_formatter/format?1583745720082 (172.16.3.90) 3.31ms referer=http://radio:8888/lab?
[W 10:22:05.837 LabApp] 422 POST /jupyterlab_code_formatter/format?1583745726553 (172.16.3.90) 1.37ms referer=http://radio:8888/lab?
```

**Comment by ryantam626** [comment]
If you open up your chrome console, you should see some error message saying you have version mismatch in the plugin, I have made this error a lot more prominent but have not released a version with that change yet.

You need to install a matching version of the lab extension.

Do `jupyter labextension install @ryantam626/jupyterlab_code_formatter@1.2.0`

**Comment by michaelaye** [comment]
Mine seem to be matching:

```bash
$ pip show jupyterlab_code_formatter                                                         (py37)
Name: jupyterlab-code-formatter
Version: 1.2.0
Summary: Server extension to power `@ryantam626/jupyterlab_code_formatter` npm package.
Home-page: https://github.com/ryantam626/jupyterlab_code_formatter
Author: Ryan Tam
Author-email: ryantam626@gmail.com
License: MIT
Location: /Users/klay6683/miniconda3/envs/py37/lib/python3.7/site-packages
Requires: packaging, notebook
Required-by:
```
```bash
$ jupyter labextension list                                                                  (py37)
JupyterLab v2.0.1
Known labextensions:
   app dir: /Users/klay6683/miniconda3/envs/py37/share/jupyter/lab
        @jupyter-widgets/jupyterlab-manager v2.0.0  enabled  OK
        @jupyterlab/geojson-extension v2.0.0  enabled  OK
        @jupyterlab/github v2.0.0  enabled  OK
        @ryantam626/jupyterlab_code_formatter v1.2.0  enabled  OK
        dask-labextension v2.0.0  enabled  OK
        ipyevents v1.7.0  enabled  OK
        itkwidgets v0.26.0  enabled  OK
        jupyter-matplotlib v0.7.2  enabled  OK
```
```bash
$ jupyter serverextension list                                                               (py37)
config dir: /Users/klay6683/.jupyter
    jupyterlab_github  enabled
    - Validating...
      jupyterlab_github 1.0.0 OK
    jupyter_nbextensions_configurator  enabled
    - Validating...
      jupyter_nbextensions_configurator 0.4.1 OK
    jupyterlab_git  enabled
    - Validating...
      jupyterlab_git  OK
    nbresuse  enabled
    - Validating...
      nbresuse  OK
config dir: /Users/klay6683/miniconda3/envs/py37/etc/jupyter
    dask_labextension  enabled
    - Validating...
      dask_labextension 2.0.0 OK
    ipyparallel.nbextension  enabled
    - Validating...
      ipyparallel.nbextension  OK
    jupyter_server_proxy  enabled
    - Validating...
      jupyter_server_proxy  OK
    jupyterlab  enabled
    - Validating...
      jupyterlab 2.0.1 OK
    jupyterlab_git  enabled
    - Validating...
      jupyterlab_git  OK
    jupyterlab_github  enabled
    - Validating...
      jupyterlab_github 1.0.0 OK
    jupyterlab_hdf  enabled
    - Validating...
      jupyterlab_hdf 0.3.0 OK
    nbdime  enabled
    - Validating...
      nbdime 1.1.0 OK
    nbresuse  enabled
    - Validating...
      nbresuse  OK
    jupyter_nbextensions_configurator  enabled
    - Validating...
      jupyter_nbextensions_configurator 0.4.1 OK
```
but I get this error in the javascript console when trying to format a cell:

https://gist.github.com/06ff135d8c799ce39678c1cf3b12e953

**Comment by ryantam626** [comment]
@michaelaye Did you do `jupyter serverextension enable --py jupyterlab_code_formatter`?

It's missing from `jupyter serverextension list`

**Comment by michaelaye** [comment]
Aha, I'm not used to do that anymore, as all my other labextensions do that for me, in the case they need a server-extension.

After executing the above, I now get these errors in the jupyterlab console:

```bash
Generating grammar tables from /Users/klay6683/miniconda3/envs/py37/lib/python3.7/site-packages/blib2to3/Grammar.txt
Writing grammar tables to /Users/klay6683/Library/Caches/black/19.10b0/Grammar3.7.6.final.0.pickle
Writing failed: [Errno 2] No such file or directory: '/Users/klay6683/Library/Caches/black/19.10b0/tmpq4kmzm3k'
Generating grammar tables from /Users/klay6683/miniconda3/envs/py37/lib/python3.7/site-packages/blib2to3/PatternGrammar.txt
Writing grammar tables to /Users/klay6683/Library/Caches/black/19.10b0/PatternGrammar3.7.6.final.0.pickle
Writing failed: [Errno 2] No such file or directory: '/Users/klay6683/Library/Caches/black/19.10b0/tmplv377r6c'
-
```

**Comment by michaelaye** [comment]
hm, looks like i only got these errors once? Functionality is there now, thanks!

**Comment by Gonzalo933** [comment]
> If you open up your chrome console, you should see some error message saying you have version mismatch in the plugin, I have made this error a lot more prominent but have not released a version with that change yet.
> 
> You need to install a matching version of the lab extension.
> 
> Do `jupyter labextension install @ryantam626/jupyterlab_code_formatter@1.2.0`

It seems I cannot install that version:

```
$ jupyter labextension install @ryantam626/jupyterlab_code_formatter@1.2.0
An error occured.
ValueError:
"@ryantam626/jupyterlab_code_formatter@1.2.0" is not compatible with the current JupyterLab
Conflicting Dependencies:
JupyterLab              Extension      Package
>=1.0.1 <2.0.0          >=2.0.0 <3.0.0 @jupyterlab/application
>=1.0.1 <2.0.0          >=2.0.0 <3.0.0 @jupyterlab/fileeditor
>=1.0.1 <2.0.0          >=2.0.0 <3.0.0 @jupyterlab/notebook
See the log file for details:  /tmp/jupyterlab-debug-15qlrtvv.log
```

**Comment by Gonzalo933** [comment]
I updated Jupyter lab and now it works.

`$ pip install jupyterlab==2.0.1`

**Comment by ryantam626** [comment]
> Aha, I'm not used to do that anymore, as all my other labextensions do that for me, in the case they need a server-extension.
> 
> After executing the above, I now get these errors in the jupyterlab console:
> 
> ```shell
> Generating grammar tables from /Users/klay6683/miniconda3/envs/py37/lib/python3.7/site-packages/blib2to3/Grammar.txt
> Writing grammar tables to /Users/klay6683/Library/Caches/black/19.10b0/Grammar3.7.6.final.0.pickle
> Writing failed: [Errno 2] No such file or directory: '/Users/klay6683/Library/Caches/black/19.10b0/tmpq4kmzm3k'
> Generating grammar tables from /Users/klay6683/miniconda3/envs/py37/lib/python3.7/site-packages/blib2to3/PatternGrammar.txt
> Writing grammar tables to /Users/klay6683/Library/Caches/black/19.10b0/PatternGrammar3.7.6.final.0.pickle
> Writing failed: [Errno 2] No such file or directory: '/Users/klay6683/Library/Caches/black/19.10b0/tmplv377r6c'
> -
> ```

I still don't quite understand what cause that error, but it's safe to ignore if it's working, it looks like black trying to initialise for the first time.

As for your comment on other plugin doing the server extension enabling, yes it's a known issue but I still haven't got around to investigate :disappointed:

**Comment by ryantam626** [comment]
@Gonzalo933 Updating jupyterlab is probably the way to go forward, but if you can't do that (say there is another plugin you really need don't support it yet) you can instead synchronise the server plugin and lab plugin vesion by first doing
1. `jupyter labextension install @ryantam626/jupyterlab_code_formatter` <- yarn will resolve the latest compatible version hopefully
2. `pip install jupyterlab_code_formatter==x.x.x` where `x.x.x` is the version step 1. installed
3. `jupyter serverextension enable --py jupyterlab_code_formatter` (not really sure if needed but doesn't hurt to do it again)

**Comment by volklor** [comment]
Hello, 

I´ve got the same problem. I have to stick with jupyterlab=1.2.6. Therefore I have to use version 1.1.0 of jupyterlab_code_formatter.

Doing the installation steps explained above, the plugin still won´t work. 

> jupyter serverextension list
> config dir: /home/ubuntu/.jupyter
>     jupyterlab_code_formatter  enabled 
>     - Validating...
>       jupyterlab_code_formatter  OK
> config dir: /home/ubuntu/miniconda3/envs/work/etc/jupyter
>     jupyter_lsp  enabled 
>     - Validating...
>       jupyter_lsp 0.7.0 OK
>     jupyterlab  enabled 
>     - Validating...
>       jupyterlab 1.2.6 OK
>     jupyterlab_hdf  enabled 
>     - Validating...
>       jupyterlab_hdf 0.3.0 OK
>     jupyterlab_s3_browser  enabled 
>     - Validating...
>       jupyterlab_s3_browser  OK
>     nbresuse  enabled 
>     - Validating...
>       nbresuse  OK
>     voila.server_extension  enabled 
>     - Validating...
>       voila.server_extension  OK

> jupyter labextension list
> JupyterLab v1.2.6
> Known labextensions:
>    app dir: /home/ubuntu/miniconda3/envs/work/share/jupyter/lab
>         @jupyter-voila/jupyterlab-preview v0.1.3  enabled  OK
>         @jupyter-widgets/jupyterlab-manager v1.1.0  enabled  OK
>         @jupyterlab/toc v2.0.0  enabled  OK
>         @krassowski/jupyterlab-lsp v0.7.1  enabled  OK
>         @pyviz/jupyterlab_pyviz v0.8.0  enabled  OK
>         @ryantam626/jupyterlab_code_formatter v1.1.0  enabled  OK
> 
> Uninstalled core extensions:
>     @pyviz/jupyterlab_pyviz
>     @ryantam626/jupyterlab_code_formatter
>     jupyterlab-logout
>     jupyterlab-s3-browser
>     jupyterlab-theme-toggle
>     jupyterlab-topbar-text

**Comment by ryantam626** [comment]
Hi, I gave an alternative here. Sorry can't copy and paste right now, on the subway.

https://github.com/ryantam626/jupyterlab_code_formatter/issues/107#issuecomment-596976760

**Comment by volklor** [comment]
I had already install `jupyterlab_code_formatter==1.1.0`.  Still not working.

However, I am wondering, why  the config dir for `jupyter_lab_formatter` is different to all the other installed extensions.

```
config dir: /home/ubuntu/.jupyter
jupyterlab_code_formatter enabled
- Validating...
jupyterlab_code_formatter OK
```

```
config dir: /home/ubuntu/miniconda3/envs/work/etc/jupyter
jupyter_lsp enabled
- Validating...
jupyter_lsp 0.7.0 OK
jupyterlab enabled
- Validating...
jupyterlab 1.2.6 OK
jupyterlab_hdf enabled
- Validating...
jupyterlab_hdf 0.3.0 OK
jupyterlab_s3_browser enabled
- Validating...
jupyterlab_s3_browser OK
nbresuse enabled
- Validating...
nbresuse OK
voila.server_extension enabled
- Validating...
voila.server_extension OK
```

**Comment by ryantam626** [comment]
> I had already install `jupyterlab_code_formatter==1.1.0`. Still not working.
> 
> However, I am wondering, why the config dir for `jupyter_lab_formatter` is different to all the other installed extensions.
> 
> ```
> config dir: /home/ubuntu/.jupyter
> jupyterlab_code_formatter enabled
> - Validating...
> jupyterlab_code_formatter OK
> ```
> 
> ```
> config dir: /home/ubuntu/miniconda3/envs/work/etc/jupyter
> jupyter_lsp enabled
> - Validating...
> jupyter_lsp 0.7.0 OK
> jupyterlab enabled
> - Validating...
> jupyterlab 1.2.6 OK
> jupyterlab_hdf enabled
> - Validating...
> jupyterlab_hdf 0.3.0 OK
> jupyterlab_s3_browser enabled
> - Validating...
> jupyterlab_s3_browser OK
> nbresuse enabled
> - Validating...
> nbresuse OK
> voila.server_extension enabled
> - Validating...
> voila.server_extension OK
> ```

not sure about why it's a different config dir..

have you restarted your jupyterlab? and server plugin changes requires a server restart

**Comment by volklor** [comment]
> > I had already install `jupyterlab_code_formatter==1.1.0`. Still not working.
> > However, I am wondering, why the config dir for `jupyter_lab_formatter` is different to all the other installed extensions.
> > ```
> > config dir: /home/ubuntu/.jupyter
> > jupyterlab_code_formatter enabled
> > - Validating...
> > jupyterlab_code_formatter OK
> > ```
> > 
> > 
> > ```
> > config dir: /home/ubuntu/miniconda3/envs/work/etc/jupyter
> > jupyter_lsp enabled
> > - Validating...
> > jupyter_lsp 0.7.0 OK
> > jupyterlab enabled
> > - Validating...
> > jupyterlab 1.2.6 OK
> > jupyterlab_hdf enabled
> > - Validating...
> > jupyterlab_hdf 0.3.0 OK
> > jupyterlab_s3_browser enabled
> > - Validating...
> > jupyterlab_s3_browser OK
> > nbresuse enabled
> > - Validating...
> > nbresuse OK
> > voila.server_extension enabled
> > - Validating...
> > voila.server_extension OK
> > ```
> 
> not sure about why it's a different config dir..
> 
> have you restarted your jupyterlab? any server plugin changes requires a server restart

Yes I have already restarted jupyterlab.

**Comment by ryantam626** [comment]
@volklor Could you describe what are you seeing in your terminal (where the jupyterlab was started) and also your web browser's console?

**Comment by volklor** [comment]
> @volklor Could you describe what are you seeing in your terminal (where the jupyterlab was started) and also your web browser's console?

I don´t know, if there is any jupyterlab_code_formatter related output, as I am not able to use `jupyterlab_code_formatter`, as there is no icon in the jupyterlabs toolbar.

```
>>> jupyter lab
[W 12:30:07.800 LabApp] Failed to fetch commands from language server spec finder`bash-language-server`:
    expected str, bytes or os.PathLike object, not tuple
[W 12:30:07.805 LabApp] Failed to fetch commands from language server spec finder`dockerfile-language-server-nodejs`:
    expected str, bytes or os.PathLike object, not tuple
[W 12:30:07.810 LabApp] Failed to fetch commands from language server spec finder`javascript-typescript-langserver`:
    expected str, bytes or os.PathLike object, not tuple
[W 12:30:07.827 LabApp] Failed to fetch commands from language server spec finder`unified-language-server`:
    expected str, bytes or os.PathLike object, not tuple
[W 12:30:07.832 LabApp] Failed to fetch commands from language server spec finder`vscode-css-languageserver-bin`:
    expected str, bytes or os.PathLike object, not tuple
[W 12:30:07.837 LabApp] Failed to fetch commands from language server spec finder`vscode-html-languageserver-bin`:
    expected str, bytes or os.PathLike object, not tuple
[W 12:30:07.842 LabApp] Failed to fetch commands from language server spec finder`vscode-json-languageserver-bin`:
    expected str, bytes or os.PathLike object, not tuple
[W 12:30:07.847 LabApp] Failed to fetch commands from language server spec finder`yaml-language-server`:
    expected str, bytes or os.PathLike object, not tuple
[I 12:30:08.167 LabApp] JupyterLab extension loaded from /home/ubuntu/miniconda3/envs/work/lib/python3.7/site-packages/jupyterlab
[I 12:30:08.167 LabApp] JupyterLab application directory is /home/ubuntu/miniconda3/envs/work/share/jupyter/lab
[I 12:30:08.759 LabApp] Serving notebooks from local directory: /home/ubuntu
[I 12:30:08.759 LabApp] The Jupyter Notebook is running at:
[I 12:30:08.759 LabApp] http://localhost:8888/?token=3e5c2cc9dbd09a0d795565867764f6493532f3ae400654e5
[I 12:30:08.759 LabApp]  or http://127.0.0.1:8888/?token=3e5c2cc9dbd09a0d795565867764f6493532f3ae400654e5
[I 12:30:08.759 LabApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[W 12:30:08.764 LabApp] No web browser found: could not locate runnable browser.
[C 12:30:08.764 LabApp] 
    
    To access the notebook, open this file in a browser:
        file:///home/ubuntu/.local/share/jupyter/runtime/nbserver-13746-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/?token=3e5c2cc9dbd09a0d795565867764f6493532f3ae400654e5
     or http://127.0.0.1:8888/?token=3e5c2cc9dbd09a0d795565867764f6493532f3ae400654e5
[I 12:30:16.749 LabApp] 302 GET /?token=3e5c2cc9dbd09a0d795565867764f6493532f3ae400654e5 (127.0.0.1) 0.89ms
[I 12:30:18.431 LabApp] Build is up to date
/[I 12:30:20.175 LabApp] Kernel started: fadc10c5-c147-4713-849d-16f1c6201a75
```

**Comment by ryantam626** [comment]
Did you install the lab extension at all?
`jupyter labextension install @ryantam626/jupyterlab_code_formatter@1.1.0`

**Comment by legout** [comment]
> Did you install the lab extension at all?
> `jupyter labextension install @ryantam626/jupyterlab_code_formatter@1.1.0`

Yes.

**Comment by volklor** [comment]
> Did you install the lab extension at all?
> `jupyter labextension install @ryantam626/jupyterlab_code_formatter@1.1.0`

Yes.

**Comment by ryantam626** [comment]
> > Did you install the lab extension at all?
> > `jupyter labextension install @ryantam626/jupyterlab_code_formatter@1.1.0`
> 
> Yes.

Nothing on your web browser's console?

**Comment by jrzech** [comment]
Hi I'm having a similar issue on my machine. I just installed it using the conda instructions from the docs, but I don't see the code formatter on the left panel and pressing the code formatter button doesn't make changes to cells.

**Comment by ryantam626** [comment]
> Hi I'm having a similar issue on my machine. I just installed it using the conda instructions from the docs, but I don't see the code formatter on the left panel and pressing the code formatter button doesn't make changes to cells.

Hi @jrzech please open a new issue and attach the diagnostic commands as laid out in the issue template :pray:

**Comment by ryantam626** [comment]
I will be closing this, as the original issue is resolved. Anyone else seeing any issue please open a new issue using the provided template :pray:
